docs/articles/documentation/test-api/selecting-page-elements/examples-of-working-with-page-dom.md

```diff
@@ -0,0 +1,132 @@
+
+## Accessing Page Element Properties
+
+To work with page elements, use TestCafe [selectors](selectors/README.md).
+Import the `Selector` function and call it passing a CSS selector inside.
+This function creates a selector object [whose API](dom-node-state.md) exposes the most used members of HTML element API.
+
+Note that all property getters are asynchronous, so add the `await` keyword.
+
+```js
+import { Selector } from 'testcafe';
+
+fixture `My fixture`
+    .page `https://devexpress.github.io/testcafe/example/`;
+
+test('My test', async t => {
+    const element     = Selector('#developer-name');
+    const clientWidth = await element.clientWidth;
+
+    // do something wuth clientWidth
+});
+```
+
+If you need to access an element property not included in the selector's API, request it explicitly by using the [addCustomDOMProperties](selectors/extending-selectors.md) method.
+
+```js
+import { Selector } from 'testcafe';
+
+fixture `My fixture`
+    .page `https://devexpress.github.io/testcafe/example/`;
+
+test('Check Label HTML', async t => {
+    const label = Selector('label').addCustomDOMProperties({
+        innerHTML: el => el.innerHTML
+    });
+
+    await t.expect(label.innerHTML).contains('input type="checkbox" name="remote"');
+});
+```
+
+## Getting a Page Element Using Custom Logic
+
+Sometimes CSS selectors are not powerful enough to identify the required page element.
+In this instance, you can introduce a function that picks the desired element.
+
+```js
+import { Selector } from 'testcafe';
+
+fixture `My fixture`
+    .page `https://devexpress.github.io/testcafe/example/`;
+
+test('My test', async t => {
+    const checkBoxesStartingWithR = Selector(() => {
+        var checkboxes     = document.querySelectorAll('input[type=checkbox]');
+        var targetElements = [];
+
+        checkboxes.forEach(node => {
+            if(node.labels[0].textContent.startsWith(' R'))
+                targetElements.push(node);
+        });
+
+        return targetElements;
+    });
+
+    await t.click(checkBoxesStartingWithR.nth(0));
+});
+```
+
+## Checking if an Element is Available
+
+Generally speaking, introducing conditions in tests is not considered good practice because it indicates that your tests are non-deterministic.
+The tested website should guarantee that the test writer knows the page state at any moment. If it is so, you need no conditions in test code.
+
+However, in practice, things are usually a bit different. Many websites contain elements that may be invisible or non-existent at times.
+In this instance, it may be a good idea to check the element availability before taking actions on it.
+
+```js
+import { Selector } from 'testcafe';
+
+fixture `My fixture`
+    .page `https://devexpress.github.io/testcafe/example/`;
+
+test('My test', async t => {
+    const element = Selector('#developer-name');
+
+    if(await element.exists && await element.visible) {
```

Comment:
Bad case. Actions wait until element became exist and visible itself.
Or I don't right @AlexanderMoskovkin

Reply:
`Many websites contain elements that may be invisible or non-existent at times.`
There is just conditional execution of an action.

Reply:
> Bad case. 

It's not a best practice but people ask about this a lot. It's actual for production sites with some kind of popups, a/b testing etc..